ERCS/erc-7946.md

```diff
@@ -0,0 +1,541 @@
+---
+eip: 7946
+title: Unidirectional Wallet Uplink aka UWULink
+description: Unidirectional app to wallet communication for private construction of transactions
+author: Moody Salem (@moodysalem), Tina Zheng (@tinaszheng)
+discussions-to: https://ethereum-magicians.org/t/erc-7946-unidirectional-wallet-uplink-aka-uwulink/24282
+status: Draft
+type: Standards Track
+category: ERC
+created: 2025-05-10
+requires: 7702
+---
+
+## Abstract
+
+Universal Wallet Uplink (UWULink) is a protocol that allows applications (dApps) and products to request a wallet to
+make a batch of contract calls in a single atomic transaction without establishing a two-way
+connection or revealing the user's address to the requester. The protocol defines a compact binary message format using
+Protocol Buffers suitable for low-bandwidth channels such as QR codes or NFC tags. Two modes of operation are
+supported: **Static Mode**, where the request payload directly contains the list of calls, and **Programmable Mode**,
+where the payload references a contract that generates the list of calls.
+
+## Motivation
+
+dApps often require users to perform multistep interactions, such as approving a token and then executing a
+swap. Traditionally, accomplishing this has required multiple user confirmations or complex wallet connectivity.
+Recent standards like EIP-5792 and EIP-7702 introduced ways to batch multiple calls into one atomic operation via
+JSON-RPC (e.g. `wallet_sendCalls`). However, those solutions assume an active connection
```

Comment:
sendCalls does not require a `from` address and no prior connection is required

Reply:
To what interface would you send the `wallet_sendCalls` RPC? In practice you connect a wallet first, check capabilities, and use `wallet_sendCalls` only if it's available. It seems like connecting a wallet is a prerequisite to know where to send the RPC.

This would be very different from the dApp having all actions available in the unconnected state, and when you trigger an action (e.g. a swap) it could pop up a modal with the QR code to perform the action, and/or a link with the `uwulink` scheme, a la magnet link, that triggers the transaction for any extension that registered itself to handle the scheme